doc/introduction/program_capture_sharp_bits.rst

```diff
@@ -0,0 +1,384 @@
+transform QNodes in a NumPy-like way and to create their own transforms with ease. 
+Your favourite transforms will still work with program-capture enabled (including
+custom transforms), but **decorating QNodes with just ``@transform_name`` will not 
+work** and will give a vague error. Additionally, decorating QNodes with the experimental 
+:func:`~.pennylane.capture.expand_plxpr_transforms` decorator is required.
+
+Consider the following toy example, which shows a tape-based transform that shifts 
+all :class:`~.pennylane.RX`` gates to the end of a circuit.
+
+.. code-block:: python
+
+    @qml.transform
+    def shift_rx_to_end(tape):
+        """Transform that moves all RX gates to the end of the operations list."""
+        new_ops, rxs = [], []
+
+        for op in tape.operations:
+            if isinstance(op, qml.RX):
+                rxs.append(op)
+            else:
+                new_ops.append(op)
+        
+        operations = new_ops + rxs
+        new_tape = tape.copy(operations=operations)
+        return [new_tape], lambda res: res[0]
+
+Decorating with just ``@shift_rx_to_end`` will not work, and will give a vague error:
+
+.. code-block:: python
+
+    @shift_rx_to_end
+    @qml.qnode(qml.device("default.qubit", wires=1))
+    def circuit():
+        qml.RX(0.1, wires=0)
+        qml.H(wires=0)
+        return qml.state()
+
+>>> print(qml.draw(circuit)())
+...
+NotImplementedError: 
+
+A requirement for tape transforms to be compatible with program capture is to further 
+decorate QNodes with the experimental :func:`~.pennylane.capture.expand_plxpr_transforms` 
+decorator:
+
+.. code-block:: python
+
+    @qml.capture.expand_plxpr_transforms
+    @shift_rx_to_end
+    @qml.qnode(qml.device("default.qubit", wires=1))
+    def circuit():
+        qml.RX(0.1, wires=0)
+        qml.H(wires=0)
+        return qml.state()
+
+>>> print(qml.draw(circuit)())
+0: ──H──RX(0.10)─┤  State
+
+Dynamic variables and transforms
+================================
+
+Some transforms in the :module:`~.pennylane.transforms` API have bespoke plxpr implementations
+that *directly* transform jaxpr:
+
+#. :func:`~.pennylane.transforms.merge_rotations`
+#. :func:`~.pennylane.transforms.single_qubit_fusion`
+#. :func:`~.pennylane.transforms.unitary_to_rot`
+#. :func:`~.pennylane.transforms.merge_amplitude_embedding`
+#. :func:`~.pennylane.transforms.commute_controlled`
+#. :func:`~.pennylane.transforms.decompose`
+#. :func:`~.pennylane.map_wires`
+#. :func:`~.pennylane.transforms.cancel_inverses`
+
+**This section applies to any transform not listed above, including custom quantum tape
+transforms**.
+
+There are a few other restrictions on QNodes and applying transforms that do not 
+have a native program-capture implementation:
+
+#. Transforms that return multiple tapes are not backwards-compatible.
+#. Transforms that return non-trivial post-processing functions are not backwards-compatible.
```

Comment:
Best to be explicit here; rather than saying backwards compatible, best to say 
```suggestion
#. Transforms that return multiple tapes are not supported.
#. Transforms that return non-trivial post-processing functions are not supported.
```